### DISCUSSION TITLE
Dynamic tailwind.config.js using an API call

### DISCUSSION THREAD

**Discussion by remidej** [tailwindlabs/tailwindcss#2012] [Help]
I writing a tutorial to help people use a headless CMS as a website builder. I want to allow people to change the theme of their website from the CMS.

Here's the workflow I'd ideally have:

1. The CMS users sets a theme for his website (for example, chooses between the predefined "light," "dark" and "playful")
2. Via a webhook, a new SSG build is triggered on the frontend (this works already)
3. The fonts, colors, spacings and all the theme data in Tailwind changes to match the new selected theme

I'm wondering if I can update the `theme` value in `tailwind.config.js` based on the result of an API call. But I don't think I use promises when defining the tailwind config.

Do you know if there would be other ways to do this?

**Comment by jk-baseer** [comment]
@remidej Hey Remi, Did you manage to find the solution for this?

**Reply by remidej** [reply]
No I did not

**Reply by jk-baseer** [reply]
Do you mind if i ask you other alternatives you found?

**Reply by remidej** [reply]
I just gave up on this one.

The solution I would imagine is that you could use the Github API (maybe via a bot/app) to listen to a webhook and create a commit that changes the tailwind config.

**Reply by jk-baseer** [reply]
hahhah That's clever. It's a multi tenant infra, so can't touch on the code. It must be from Server. 

We are studying the feasibility of using Inline classes with emotion. or worst case find tailwind alternative but we love tailwind, so it's going to be hard.

**Reply by remidej** [reply]
It could also be a reason to fork Tailwind (and maybe submit a PR)

**Reply by alexandre-tobia** [reply]
@remidej @jk-baseer It might be old but have you concidered to use css var in your tailwind.config.js ? And with your api, you inject thoses variables into the head.

Tailwind config: 
```js
'primary-color': 'var(--primary-color)',
```

HTML Head (injected with api) : 
```html
<head>
 <styles>
      :root {
          --primary-color: "red";
       }
 </styles>
</head>
```

**Reply by jk-baseer** [reply]
Hi @alexandre-tobia 
Thanks man for the follow up. Yes we did using the variable method but not directly touching the tailwind config. we are using some sort of bridge method to do this. Are you trying a similar approach?

**Reply by alexandre-tobia** [reply]
@jk-baseer  I haven't investigated any further at the moment, but if you've got anything working on your end, I'd love to hear about it!

**Reply by jaguarfi** [reply]
At least on the daisyui website they recommend this package to switch themes. It may be worth trying it out https://github.com/saadeghi/theme-change
